public/app/features/browse-dashboards/components/BulkActions/BulkActionFailureBanner.test.tsx

```diff
@@ -0,0 +1,151 @@
+import { render, screen } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+
+import { BulkActionFailureBanner, MoveResultFailed } from './BulkActionFailureBanner';
+
+const setup = (resultOverrides?: Array<Partial<MoveResultFailed>>, onDismissOverride?: () => void) => {
+  const defaultFailedItems: MoveResultFailed[] = [
+    {
+      status: 'failed',
+      title: 'Dashboard 1',
+      errorMessage: 'Permission denied',
+    },
+    {
+      status: 'failed',
+      title: 'Dashboard 2',
+      errorMessage: 'Network error',
+    },
+  ];
+
+  const result =
+    resultOverrides !== undefined
+      ? resultOverrides.map((override) => ({ status: 'failed' as const, title: 'Default Title', ...override }))
+      : defaultFailedItems;
+
+  const onDismiss = onDismissOverride || jest.fn();
+
+  const props = {
+    result,
+    onDismiss,
+  };
+
+  return {
+    user: userEvent.setup(),
+    ...render(<BulkActionFailureBanner {...props} />),
+    props,
+  };
+};
+
+describe('BulkActionFailureBanner', () => {
+  it('should display error alert with correct item count', () => {
+    const testData = [{ title: 'Single Item', errorMessage: 'Single error' }];
+    setup(testData);
+
+    // Test that an alert is rendered
+    const alert = screen.getByRole('alert');
+    expect(alert).toBeInTheDocument();
+
+    // Test structure: should have same number of list items as input data
+    const listItems = screen.getAllByRole('listitem');
+    expect(listItems).toHaveLength(testData.length);
+  });
+
+  it('should render correct number of failed items with proper structure', () => {
+    const testData = [
+      { title: 'Failed Dashboard A', errorMessage: 'Access denied' },
+      { title: 'Failed Dashboard B', errorMessage: 'Validation failed' },
+    ];
+    setup(testData);
+
+    const alert = screen.getByRole('alert');
+    expect(alert).toBeInTheDocument();
+
+    // Test structure: number of list items matches input
+    const listItems = screen.getAllByRole('listitem');
+    expect(listItems).toHaveLength(testData.length);
+
+    // Test that each item has the expected structure (title + error message)
+    listItems.forEach((item, index) => {
+      const title = testData[index].title;
+      const errorMessage = testData[index].errorMessage;
+
+      if (title) {
+        expect(item).toHaveTextContent(title);
+      }
+      if (errorMessage) {
+        expect(item).toHaveTextContent(errorMessage);
+      }
```

Comment:
nitpick, don't need to necessarily fix it here as there are problems with this in other tests throughout the codebase:
We should avoid conditionals as they can result in ever-green tests (there is [this ESLint rule we could eventually use](https://github.com/jest-community/eslint-plugin-jest/blob/main/docs/rules/no-conditional-in-test.md))